tests/suites/main_test.function

```diff
@@ -293,6 +301,7 @@ int main(int argc, const char *argv[])
     if( pointer != NULL )
     {
         mbedtls_fprintf( stderr, "all-bits-zero is not a NULL pointer\n" );
+        platform_teardown();
```

Comment:
This function has so many exit points that makes it difficult to cleanup. It may be tempting to refactor and introduce ```goto```. However, #930 handles that. Hence, it can stay as it is.